src/com/mellanox/cicd/Matrix.groovy

```diff
@@ -239,21 +282,46 @@ def runK8(image, branchName, config, axis) {
         str += "$key = $val\n"
     }
 
-    run_shell('printf "%s"' +  '"' + str + '"', "Matrix axis parameters")
-    
-
+    config.logger.debug("runK8 | str: ${str}")
 
     def listV = parseListV(config.volumes)
     def cname = image.get("name").replaceAll("[\\.:/_]","")
-    def nodeSelector = getConfigVal(config, ['kubernetes', 'nodeSelector'], "")
-
-    podTemplate(cloud: cloudName, runAsUser: "0", runAsGroup: "0",
-                nodeSelector: nodeSelector,
-                containers: [
-                    containerTemplate(name: cname, image: image.url, ttyEnabled: true, alwaysPullImage: true, command: 'cat')
-                ],
-                volumes: listV
-                )
+
+    config.logger.debug("runK8 | arch: ${axis.arch}")
+
+    def k8sArchTable = getConfigVal(config, ['kubernetes','arch_table'], "")
+    def nodeSelector = ''
+    def jnlpImage = ''
+
+    if (!k8sArchTable) {
```

Comment:
populate k8ArchTable with defaults if it is null, similar to below and proceed with flow.
https://blog.mrhaki.com/2020/02/groovy-goodness-parse-yaml-with.html

Reply:
Fixed.